---
title: "ResearcHers Code takeover"
date: 2021-02-12T14:45:03+01:00
draft: true
toc: true
---

{{< color-block style="info" >}}
I ran the ResearcHers Twitter account from the 8th until the 12th February 2021 - the below are Tweets I copied over and slightly edited/shortened for the blog & to update the reader.
{{< /color-block >}}

## Monday: history

Hello! I'm Samantha, and I'll be tweeting from this account for the week! I'm a 3rd year PhD student at the University of Birmingham in the UK. My field is Clinical Informatics and I am looking at inflammation! Today I’ll talk about my experience learning to code throughout education: when it started, how it progressed, and what I can do now!

I was lucky to attend a primary school with a computer suite. My first search engine was Ask Jeeves. I didn't understand what a search engine was at the time and seeing "Ask" meant I literally asked it full questions! These days Google already knows what I want to ask before I have finished typing.

The first time I encountered coding was when I was around 8 years old. A school visitor introduced us to "drag & drop" visual programming with a tool called Flowol (somewhat like Scratch). 
We started with a few exercises: shown how to drag and drop these shapes and that they make a sequence of commands.
I remember enjoying it so much that I rushed through to ensure I could try all the exercises.

We finished with a competition: we were grouped up and it was a race to solve the traffic light problem and having to consider the rules.
I remember my partner and I struggled but I didn't want to stop and so I kept working on. I remember my sequence of shapes became very complex at the end with arrows pointing everywhere, but it worked!

Our school also had a programmable floor robot. There was only one so we couldn't individually try but the teacher would show us how to run it. 
As much as I enjoyed these experiences I wouldn't cross paths with programming again until I was in college.

In college (aged 16) I decided to study IT as I enjoyed it and realised I was actually pretty good at it too.
I originally wanted to be a midwife and undertook child development and health and social care studies alongside IT. I wondered maybe there would be a way I could somehow combine healthcare and IT in my future. 

My IT course at college covered many areas, including: animation, business, and web development. I even have the animation project I made!
******************************

I really enjoyed the web modules. I remember our first task was to build a basic website with `HTML` and `CSS` with Adobe Dreamweaver. I still didn't have a computer at home, so I spent my evenings at school to make sure I wouldn't fall behind.
The web module is where I also met `Javascript` - although the course was not programming-focused, the teacher provided code snippets which I further explored.
When I got a laptop of my own, I tried to learn `Javascript` in my spare time. 

It was at this stage that I had to start thinking about my future: what were my career options? Did I actually want to be a midwife? Did I want to focus on IT? I found IT much easier and it was so fun. I asked my IT teacher for advice… My IT teacher strongly encouraged me to apply for computer science and to choose a university outside my hometown to gain independence and confidence.


I started my undergraduate degree ************** link to undergrad blog - at Aberystwyth University, UK, studying Computer Science in 2014. Some coding skills I learnt in my first year included: `Java`, `Arduino C`, and my old friend `Javascript`.
The CompSci department runs an activity weekend every year for all first years for various team building activities - without phone signal!

One of my first challenges as an undergrad was the introduction to Linux and the command line interface. It was nothing like I've ever done before. I stepped into this whole new computing world that included other operating systems! I had previously only ever interacted with a Windows OS.

I enjoyed - and struggled - with the `Arduino`. On the other hand, I failed `Java`: going from `Arduino C` to object-orientated `Java` was too confusing for me.
I found it hard to wrap my head around all these new concepts, I tried taking out a book or two from the library but I didn't understand it as much as I had hoped.

Due to these struggles and other great advice: it was recommended to swap over to Business IT for my second year onward, and I was happy with that decision. I found programming languages difficult yet enjoyed web building. But it was not the end of my coding experiences!

My second year modules included: web programming, system administration, and databases.
During web programming I learnt `PHP` and in databases I learnt `PostgreSQL`.
I also chose some modules outside the department as I wanted to widen my career choices.

My third year included a group project and my final dissertation. My final dissertation was creating a website for a small cake business with a database and a forum for users to talk recipes.
This is also the year I was introduced to Machine Learning. The machine learning module focused on using the Weka software.
At the end of my third year, I needed to start thinking about my next steps: I had applied for Data Science.

***************** link to bioinformatics & postgrad blogs

Over the Summer of 2017 - post undergraduate - with some help I started to learn other skills, specifically `Python`.
And then programming **clicked**. Despite my prevoius troubles, I finally felt like I understood a programming language with ease! 

Autumn 2017 was the start of my Masters in Data Science! I explored various topics: statistics; and reintroduced in machine learning and databases. I also learnt new skills: `R` and `MongoDB`.
One of my most memorable university modules was the introduction to `Python`. This was taught via lectures and practicals, both with live coding during the session and the lecturer would get us involved.
My Masters project used a varity of skills, especially `Python` looking at Acidobacteria and its DNA content - as part of this I made my first `Python` tool and released it on GitHub.

**********************************

## Tuesday: PhD

The first thing I wanted to discuss is PhD applications! I’ve had a few questions from Masters students over the past year about PhD application essays & interviews - I’ll write some tips from the blog in this thread!
1. Skills to highlight: mention all the modules you did during your undergraduate and/or masters! All are relevant, but maybe focus a little more on the ones that might be specifically related to the PhD project! Also what skills have you USED?
2. Don’t worry if you don’t think you’re good at a particular skill - what’s important is that you know about it and doing the PhD gives you the chance to improve that skill!
3. Don’t forget the non-technical skills like “communication” or “presenting” - you’ll probably have to present your PhD work at some point!
4. If you are presenting at the interview, put skills or buzzwords in bold to highlight what you can do! Be prepared for possible questions like, “how would you do it differently?” - but questions about the presentation depend on the topic
5. Interview questions: they can range from technical, or questions about yourself! Be prepared for the dreaded question: “what is your biggest weakness?”
6. A personal opinion is that I felt like the interview panel wanted to get to know me: was I a good fit? Did I know what I was talking about?
7. Technical questions can include, “how would you handle a particular dataset?” or “how would you run a GWAS?” - remember, some questions have no wrong answers - they genuinely just want to know your response!
8. Try and throw in some buzzwords that may be relevant for the PhD in terms of skill! But of course make sure you know what these words are - they may pick up if you start throwing random words around
9. Do some background research into the PhD supervisor! Find out their main paper topics, or what people in their lab are doing
10. The PhD is a learning and development journey, so even if you don't fully understand some things try not to worry: the PhD is a time for the supervisor to pass on their knowledge and you’ll learn other stuff
11. Be yourself! Don’t be afraid to answer a question in your own way! And don’t be afraid if you don’t have an answer!

My PhD is focused on inflammation! I am looking into non-traditional biomarkers of inflammation through various methods & data sources: structured electronic health records and unstructured clinical letters. Some possible non-traditional biomarkers could include blood counts/biochemistry or clinical letters telling us a patients’ current status, treatment, and symptoms - additionally some associated genetic traits.
My original PhD title was different: the plan was to use various Biobanks studies and link up missing gaps, however we soon learnt that the data wasn’t available... The new aim is to work on patients in the UK Biobank, which is a long study aimed to investigate the contributions and development of a disease both genetically and environmentally.

"Creating an AI based data assistant to bridge genotype to metadata linking primary clinical data to biobank sample" - my original PhD title

To link back to yesterday and my coding experience - currently my main language is Python, sometimes I use R for statistics or plotting. At first, I struggled a lot with Python dictionaries but I’ve finally become friends with them.

## Wednesday: skills

My main interests and career ambitions are: machine learning, ontologies, and natural language processing!

Machine learning (ML) is “learning from data”, it can be supervised prediction with labeled data or unsupervised clustering finding underlying relationships in datasets. I’m personally interested in clustering as I like to observe patterns and in my data and try to interpret the cluster groups, I’ve used K-means, hierarchical, DBScan, plus more! Because I work with such large datasets, I have used various dimensionality reduction methods: specifically t-SNE and PCA. An interesting tool I found is PCAmixdata for R - essentially it combines a PCA (continuous) and MCA (categorical) results - I would definitely recommend! The main toolkit I use for my clustering pipelines is the scikit-learn Python module! I use it for dimensionality reduction, optimal number of K, and clustering! I like to say that there’s no “right way” to do machine learning - but parameter finetuning can improve results! Don’t forget visualisation - it’s important to see what the data looks like too.

Ontologies condense a domain of knowledge in formalised structure. The concepts of an ontology have metadata and relationships, for example in human anatomy: hand “part of” arm [synonym = upper limb]. Ontologies allow us to do association text mining: we can extract important sentences from a document using terms and their corresponding synonyms. Linking with natural language processing (NLP), NLP is using computers to look at human language and interaction. I aim to look at those clinical letters of patients with inflammation! So far I have looked into word vectorising (word2vec): looking how terms are closely related to one-another from different sources, combining this with ontologies we can do semanity similarity! If you want a way to visualise an ontology, I recommend: WebVOWL - it’s an online application with interactive features! If you are interested in NLP, I recommend: spaCy - it supports many languages, very fast, and has pre-trained models! I personally find this tool the best one I’ve used for PoS tagging and such thus far.
Doing my work for association text mining or semantic similarity with documents and ontologies is difficult because a lot of tools that exist for NLP tasks don’t include features for ontology extraction... A first step to overcome this barrier, I made Jabberwocky! An initial plan to plug-in an ontology for associated text mining: using an ontology term and it collects the synonyms - my future plan is to incorporate spaCy with it!

Finally, currently I have been doing some genome-wide association studies (GWAS): looking into genetic variants of a group of people to observe if any variant is associated with a trait. An example of this can be looking at patients with diabetes and observing if they have the same variant which is associated with a particular trait. To be quite honest, this is not my strongest point - as I don’t feel quite knowledgeable in the genetics area yet, but individuals in my lab co-created an R & bash pipeline which they shared with me!

## Thursday: struggles

One of my biggest struggles is trying to explain my methods or results to others - I’m still learning about terms I should be using or still trying to interpret the biological meaning... An example of this is: using “feature” instead of “column” or “variable” when describing my data or methods.
Writing - I actually don’t mind writing even though my grammar does seem to be terrible...BUT it doesn’t stop me from trying, and I appreciate that others spend time improving my work! It started off quite difficult to accept feedback, I felt like I was so bad at writing and I was embarrassed - but I’ve learnt to accept and value comments! I appreciate those who help me to improve my work.
It’s important for me to have some sort of IDE while programming. I like to easily run a script and observe my variables - I need to “see” my output before I can continue, maybe it’s because I’m still not 100% confident in my programming skills. I use Spyder for Python! But to run lines separately, some good tools are Jupyter Notebook (Python) or RStudio (R).
Conferences: I’m quite nervous about conferences because I worry that I’ll get asked a question about my work which I can’t answer - but I know they’re good for networking and career experiences!

Mental health! It’s very important to take time for yourself and do things you enjoy! I’ve had to take a day or few off in the past year because of the overwhelming stress… No matter if you’re a student, in academia, or industry, you need to take care of yourself and it’s important for businesses/companies to take care of their employees! In 2018 I wrote a blog post about witnessing my partner writing their PhD: watching them sad, happy, angry…it was hard to watch that - and now being in that position, I completely understand!
Trying to survive this pandemic has been difficult. I’m lucky to not have my work affected as it’s computational - but I still struggle: my body aches as I don’t get my daily walk into work & my back hurts from my cheap chair! I’m too nervous to go outside for a walk because I see a lot of people not wearing masks and these variants scare me...

## Friday: funday

Hobbies! Since the pandemic I’ve started a bunch of hobbies: knitting, clay sculpting, and even coding for fun! I have a Colours project on GitHub, I make small simple tools with the techniques I’ve learnt throughout my PhD! Music I listen to while I work: I love movie soundtracks! Such as Tron Legacy & Interstellar - Hans Zimmer is my go-to!